### PR TITLE
[PHP 7.4] Add array spread

### PR DESCRIPTION
**Pull request by TomasVotruba**
Based on RFC: https://wiki.php.net/rfc/spread_operator_for_array

Based on PR: https://github.com/php/php-src/pull/3640/files

@nikic There you go :)

### CODE REVIEW

File: lib/PhpParser/Node/Expr/ArrayItem.php [hunk 1]
```diff
@@ -21,17 +23,18 @@ class ArrayItem extends Expr
      * @param bool      $byRef      Whether to assign by reference
      * @param array     $attributes Additional attributes
      */
-    public function __construct(Expr $value, Expr $key = null, bool $byRef = false, array $attributes = []) {
+    public function __construct(Expr $value, Expr $key = null, bool $byRef = false, array $attributes = [], bool $unpack = false) {
```
Comment (TomasVotruba) [inline lib/PhpParser/Node/Expr/ArrayItem.php]: For BC as last argument

File: test/code/formatPreservation/array_spread.test [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+$items = [
+...$value
+];
+-----
+$array = $stmts[0]->expr;
+$array->items[] = new Expr\ArrayItem(new Expr\Variable('b'));
+-----
+<?php
+$items = [
+...$value, $b
```
Comment (TomasVotruba) [inline test/code/formatPreservation/array_spread.test]: I have no idea why the `$b` is not added

File: test/code/formatPreservation/array_spread.test [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Array spread
+-----
+<?php
+$items = [
+...$value
+];
+-----
+$array = $stmts[0]->expr;
```
Comment (nikic) [inline test/code/formatPreservation/array_spread.test]:
```suggestion
$array = $stmts[0]->expr->expr;
```

**Comment by TomasVotruba** [comment]
Should this be reflected in `ArrayItem` constructor?

```php
$arr1 = [1, 2, 3];
$arr2 = [...&$arr1]; //invalid syntax
```

**Comment by nikic** [comment]
@TomasVotruba Nah, we don't check AST validity in ctors.

**Comment by TomasVotruba** [comment]
Ready for merge then

**Comment by TomasVotruba** [comment]
:+1:

**Comment by TomasVotruba** [comment]
Hi, any ETA for next patch release that will include this and arrow function?

Rector already supports the migration and I'd like to test it in advance to prevent any missed edge-cases.

**Comment by TomasVotruba** [comment]
@nikic Thanks for the release :tada:
